progenitor-impl/src/method.rs

```diff
@@ -710,9 +710,12 @@ impl Generator {
             let step_params = method.params.iter().map(|param| {
                 if param.api_name.as_str() == "page_token" {
                     quote! { state.as_deref() }
-                } else if let OperationParameterKind::Query(_) = param.kind {
-                    // Query parameters are None; having page_token as Some(_)
-                    // is mutually exclusive with other query parameters.
+                } else if param.api_name.as_str() != "limit"
```

Comment:
This is the change I'm least sure about, but I _think_ it's fine? We should only be here if we have a non-`None` `dropshot_pagination`, and `dropshot_pagination_data` explicitly checks for both `"page_token"` and `"limit"` as query parameters.

Reply:
Yeah, this is the part I'm least sure about as well. My intention (unexpressed in any comments) is that the consumer might specify the size of the first page to optimize for latency and then rely on the server to give us chunky, throughput oriented pages subsequently. Imagine, for example, a CLI where we  want to start printing output quickly.

Alternatively, would it make sense to have parameters first the first page and other pages?

In either case, if we don't include this change we might consider updating the function parameters docs which, currently, come from dropshot via the OpenAPI doc.

Reply:
> Alternatively, would it make sense to have parameters for the first page and other pages?

I thought about this. It doesn't seem unreasonable to me, although I suspect most clients _ought_ to specify `Some(small_value)` for the first page and `None (server choice)` for the other pages, right? I wonder if it would help to put this in a new type specifically for these parameters for clarity? Totally spitballing, something like:

```rust
pub struct StreamPaginationLimits {
    // or just omit this one and recommend clients use `.take(_)`
    pub total: Option<NonZeroU32>,
    pub first_page: Option<NonZeroU32>,
    pub remaining_pages: Option<NonZeroU32>,
}
```

with helper methods for "let the server pick all the limits, I don't care" and "I just want a small first page, let the server pick for remaining pages".

Reply:
Another thought: I'm not sure the stream interface actually does the "small first page, larger subsequent pages" thing you want, although I guess it depends on the client. I _think_ if you're using

```rust
while let Some(item) = stream.next() { .. }
```

with a first page limit of 10 and a subsequent page limit of 100, you'll incur the latency for the `GET (10)` on the first `nex()`, the next 9 will return immediately, then you'll incur the latency for `GET (100)` on the 11th call to `next()`, right?